server/e2e/publicapi_test.go

```diff
@@ -370,11 +371,97 @@ func TestPublicAPI(t *testing.T) {
 		IsEqual(map[string]any{
 			"error": "not found",
 		})
+
+	// make the project limited
+	prj.Publication().SetScope(project.PublicationScopeLimited)
+	prj.Publication().SetAssetPublic(true)
+	prj.Publication().GenerateToken()
+	token := prj.Publication().Token()
+	lo.Must0(repos.Project.Save(ctx, prj))
+
+	// invalid token
+	e.GET("/api/p/{project}/{model}", publicAPIProjectAlias, publicAPIModelKey).
+		WithHeader("Origin", "https://example.com").
+		WithHeader("Authorization", "secret_abc").
+		WithHeader("Content-Type", "application/json").
+		Expect().
+		Status(http.StatusUnauthorized)
+
+	// valid token
```

Comment:
try to access another project within the same workspace, it should be unauthorized